espressostreamer/utils.go

```diff
@@ -1,35 +1,46 @@
 package espressostreamer
 
+const (
+	FilterAndFind_Remove = iota
+	FilterAndFind_Keep
+	FilterAndFind_Target
+)
+
 // FilterAndFind filters an array in-place and returns the matching element based on a comparison function.
 // The comparison function should return:
-//   - 0 for the element to be returned
-//   - negative number for elements to be removed
-//   - positive number for elements to be kept
+//   - FilterAndFindTarget for the element to be returned, will be kept in the array
+//   - FilterAndFindKeep for elements to be kept
+//   - FilterAndFindRemove for elements to be removed
 //
-// Returns the found element (if any) and a boolean indicating if an element was found.
-func FilterAndFind[T any](arr *[]T, compareFunc func(T) int) (T, bool) {
-	var found T
+// Returns the index of the found element (if any)
+func FilterAndFind[T any](arr *[]T, compareFunc func(T) int) int {
+
 	var hasFound bool
+	idx := -1
 
 	if arr == nil || len(*arr) == 0 {
-		return found, false
+		return idx
 	}
 
 	j := 0
 	for i := 0; i < len(*arr); i++ {
 		result := compareFunc((*arr)[i])
+
+		if result == FilterAndFind_Remove || (result == FilterAndFind_Target && hasFound) {
```

Comment:
I dont understand why we need to continue if we have `hasFound` here?

Reply:
So this function not only finds the target, but also removes the duplicated and outdated ones. if it continues, the rest of the array will also be cleaned up